vmm/src/device_manager.rs

```diff
@@ -1710,7 +1711,8 @@ impl DeviceManager {
     ) -> DeviceManagerResult<Vec<(VirtioDeviceArc, bool, Option<String>)>> {
         let mut devices = Vec::new();
 
-        let mm = &self.memory_manager.lock().unwrap();
+        let mm = self.memory_manager.clone();
+        let mm = mm.lock().unwrap();
         if let (Some(region), Some(resize)) = (&mm.virtiomem_region, &mm.virtiomem_resize) {
```

Comment:
I don't see why this needed to be changed?

Reply:
I moved it to another commit but this is needed to avoid the following failure:
```
error[E0502]: cannot borrow `*self` as mutable because it is also borrowed as immutable                                                                                                                                                                              
    --> vmm/src/device_manager.rs:1761:13                                                                                                                                                                                                                            
     |                                                                                                                                                                                                                                                               
1740 |         let mm = self.memory_manager.lock().unwrap();                                                                                                                                                                                                         
     |                  ------------------- immutable borrow occurs here                                                                                                                                                                                             
...                                                                                                                                                                                                                                                                  
1761 |             self.add_migratable_device(Arc::clone(&virtio_mem_device) as Arc<Mutex<dyn Migratable>>);                                                                                                                                                         
     |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here                                                                                                                               
...                                                                                                                                                                                                                                                                  
1765 |     }                                                     
     |     - immutable borrow might be used here, when `mm` is dropped and runs the `Drop` code for type `std::sync::MutexGuard`

error: aborting due to previous error

```